mesonbuild/ast/interpreter.py

```diff
@@ -60,39 +75,136 @@
         UMinusNode,
     )
 
-class DontCareObject(MesonInterpreterObject):
-    pass
-
-class MockExecutable(MesonInterpreterObject):
-    pass
+# `IntrospectionFile` is to the `IntrospectionInterpreter` what `File` is to the normal `Interpreter`.
+@dataclass
+class IntrospectionFile(HoldableObject):
+    subdir: str
+    rel: str
 
-class MockStaticLibrary(MesonInterpreterObject):
-    pass
+    def to_abs_path(self, root_dir: Path) -> Path:
+        return (root_dir / self.subdir / self.rel).resolve()
 
-class MockSharedLibrary(MesonInterpreterObject):
-    pass
+    def __hash__(self) -> int:
+        return hash((self.__class__.__name__, self.subdir, self.rel))
 
-class MockCustomTarget(MesonInterpreterObject):
+# `IntrospectionFileHolder` is to the `IntrospectionInterpreter` what `FileHolder` is to the normal `Interpreter`.
+class IntrospectionFileHolder(ObjectHolder[IntrospectionFile]):
     pass
 
-class MockRunTarget(MesonInterpreterObject):
-    pass
-
-ADD_SOURCE = 0
-REMOVE_SOURCE = 1
-
-_T = T.TypeVar('_T')
-_V = T.TypeVar('_V')
-
+# `IntrospectionDependency` is to the `IntrospectionInterpreter` what `Dependency` is to the normal `Interpreter`.
+@dataclass
+class IntrospectionDependency(MesonInterpreterObject):
+    name: str
+    required: T.Union[bool, UnknownValue]
+    version: T.List[str]
+    has_fallback: bool
+    conditional: bool
+    node: FunctionNode
+
+# `IntrospectionBuildTarget` is to the `IntrospectionInterpreter` what `BuildTarget` is to the normal `Interpreter`.
+@dataclass
+class IntrospectionBuildTarget(MesonInterpreterObject):
+    name: str
+    machine: str
+    id: str
+    typename: str
+    defined_in: str
+    subdir: str
+    build_by_default: bool
+    installed: bool
+    outputs: T.List[str]
+    source_nodes: T.List[BaseNode]
+    extra_files: BaseNode
+    kwargs: T.Dict[str, TYPE_var]
+    node: FunctionNode
+
+def flatten_nested_lists(ar: T.Any) -> T.List[T.Any]:
```

Comment:
Couldn't you use interpreterbase.helpers.flatten instead?

Reply:
I just replaced flatten_nested_lists with flatten. I will check later if the ci is green.